src/drivers/si4063/si4063.c

```diff
@@ -218,10 +223,88 @@ void si4063_disable_tx()
     si4063_set_state(SI4063_STATE_SLEEP);
 }
 
+// Returns number of bytes sent from *data
+// If less than len, remaining bytes will need to be used to top up the buffer
+uint16_t si4063_start_tx(uint8_t *data, int len)
+{
+    // Clear TX FIFO
+    uint8_t fifo_clear_data[] = {1};
+    si4063_send_command(SI4063_COMMAND_FIFO_INFO, 1, fifo_clear_data);
+    si4063_wait_for_cts();
+
+    // Add our data to the TX FIFO
+    int fifo_len = len;
+    if(fifo_len > 64) {
+        fifo_len = 64;
+    }
+    si4063_send_command(SI4063_COMMAND_WRITE_TX_FIFO, fifo_len, data);
+
+    // Start transmitting
+    uint8_t tx_cmd[] = {
+        0, // channel
+        SI4063_STATE_SLEEP << 4,
+        len >> 8,
+        len & 0xFF,
+        0 // delay
+    };
+    si4063_send_command(SI4063_COMMAND_START_TX, sizeof(tx_cmd), tx_cmd);
+    si4063_wait_for_cts();
+
+    return fifo_len;
+}
+
+// Add additional bytes to the si4063's FIFO buffer
+// Needed for large packets that don't fit in its buffer.
+// Keep refilling it while transmitting
+// Returns number of bytes taken from *data
+// If less than len, you will need to keep calling this
+uint16_t si4063_refill_buffer(uint8_t *data, int len)
+{
+    uint8_t response[2];
+
+    // Check how many bytes we have free
+    si4063_send_command(SI4063_COMMAND_FIFO_INFO, 0, NULL);
+    si4063_read_response(sizeof(response), response);
+
+    uint8_t free_space = response[1];
```

Comment:
Very odd - this throws a warning that `response[1]` may not be initialized. It goes away if I change line 263 to `uint8_t response[3];`

Reply:
Fixed by adding a `... = {0, 0};`. Seems weird this is needed when it's not needed for `response[3]`